pkg/operator/controllerimagepruner.go

```diff
@@ -0,0 +1,384 @@
+		if err != nil {
+			return fmt.Errorf("failed to get pruner jobs: %s", err)
+		}
+
+		lastPrunerJobConditions := []batchv1.JobCondition{}
+		if len(prunerJobs) > 0 {
+			sort.Sort(ByCreationTimestamp(prunerJobs))
+			lastPrunerJobConditions = prunerJobs[len(prunerJobs)-1].Status.Conditions
+		}
+
+		c.syncPrunerStatus(pcr, prunerCronJob, lastPrunerJobConditions)
+
+		metadataChanged := strategy.Metadata(&prevPCR.ObjectMeta, &pcr.ObjectMeta)
+		specChanged := !reflect.DeepEqual(prevPCR.Spec, pcr.Spec)
+		if metadataChanged || specChanged {
+			klog.Infof("Updating pruner cr")
+			difference, err := object.DiffString(prevPCR, pcr)
+			if err != nil {
+				klog.Errorf("unable to calculate difference in %s: %s", utilObjectInfo(pcr), err)
+			}
+			klog.Infof("object changed: %s (metadata=%t, spec=%t): %s", utilObjectInfo(pcr), metadataChanged, specChanged, difference)
+
+			updatedPCR, err := c.clients.RegOp.ImageregistryV1().ImagePruners().Update(pcr)
+			if err != nil {
+				if !errors.IsConflict(err) {
+					klog.Errorf("unable to update %s: %s", utilObjectInfo(pcr), err)
+				}
+				return err
+			}
+
+			// If we updated the Status field too, we'll make one more call and we
+			// want it to succeed.
+			pcr.ResourceVersion = updatedPCR.ResourceVersion
+
+		}
+
+		pcr.Status.ObservedGeneration = pcr.Generation
+		statusChanged := !reflect.DeepEqual(prevPCR.Status, pcr.Status)
+		if statusChanged {
+			difference, err := object.DiffString(prevPCR, pcr)
+			if err != nil {
+				klog.Errorf("unable to calculate difference in %s: %s", utilObjectInfo(pcr), err)
+			}
+			klog.Infof("object changed: %s (status=%t): %s", utilObjectInfo(pcr), statusChanged, difference)
+
+			_, err = c.clients.RegOp.ImageregistryV1().ImagePruners().UpdateStatus(pcr)
+			if err != nil {
+				if !errors.IsConflict(err) {
+					klog.Errorf("unable to update status %s: %s", utilObjectInfo(pcr), err)
+				}
+				return err
+			}
+		}
+	}
+
+	if _, ok := applyError.(permanentError); !ok {
+		return applyError
+	}
+
+	return nil
+}
+
+func (c *ImagePrunerController) eventProcessor() {
+	for {
+		obj, shutdown := c.workqueue.Get()
+		if shutdown {
+			return
+		}
+
+		klog.V(1).Infof("get event from image pruner workqueue")
+		func() {
+			defer c.workqueue.Done(obj)
+
+			if _, ok := obj.(string); !ok {
+				c.workqueue.Forget(obj)
+				klog.Errorf("expected string in workqueue but got %#v", obj)
+				return
+			}
+
+			if err := c.sync(); err != nil {
+				c.workqueue.AddRateLimited(imagePrunerWorkQueueKey)
+				klog.Errorf("(imaeg pruner) unable to sync: %s, requeuing", err)
+			} else {
+				c.workqueue.Forget(obj)
+				klog.Infof("event from image pruner workqueue successfully processed")
+			}
+		}()
+	}
+}
+
+func (c *ImagePrunerController) handler() cache.ResourceEventHandlerFuncs {
+	return cache.ResourceEventHandlerFuncs{
+		AddFunc: func(o interface{}) {
+			if clusterOperator, ok := o.(*configapiv1.ClusterOperator); ok {
```

Comment:
this controller doesn't manage ClusterOperators

Reply:
Yep, this was still listed as a WIP and I am doing cleanup.
Updated in next push